slashing-protection/src/integration-test/java/tech/pegasys/web3signer/slashingprotection/WatermarkImportingIntegrationTest.java

```diff
@@ -0,0 +1,196 @@
+    slashingProtection.importData(input);
+    assertThat(getWatermark(VALIDATOR_ID))
+        .isEqualToComparingFieldByField(
+            new SigningWatermark(VALIDATOR_ID, UInt64.valueOf(4), null, null));
+  }
+
+  @Test
+  public void
+      blockWaterMarkIsUpdatedIfImportHasSmallestMinimumBlockLargerValueThanExistingMaxBlock()
+          throws JsonProcessingException {
+    final UInt64 initialBlockSlot = UInt64.valueOf(3);
+    insertValidator(PUBLIC_KEY, VALIDATOR_ID);
+    slashingProtection.registerValidators(List.of(PUBLIC_KEY));
+    insertBlockAt(initialBlockSlot, PUBLIC_KEY);
+    insertBlockAt(UInt64.valueOf(10), PUBLIC_KEY); // max = 10, watermark = 3 (As was first).
+    assertThat(getWatermark(VALIDATOR_ID).getSlot()).isEqualTo(initialBlockSlot);
+
+    final InputStream input = createInputDataWith(List.of(20, 19), emptyList());
+    slashingProtection.importData(input);
+    assertThat(getWatermark(VALIDATOR_ID).getSlot()).isEqualTo(UInt64.valueOf(19));
+  }
+
+  @Test
+  public void blockWatermarkIsNotUpdatedIfLowestBlockInImportIsLowerThanHighestBlock()
+      throws JsonProcessingException {
+    insertValidator(PUBLIC_KEY, VALIDATOR_ID);
+    slashingProtection.registerValidators(List.of(PUBLIC_KEY));
+    insertBlockAt(UInt64.valueOf(3), PUBLIC_KEY);
+    insertBlockAt(UInt64.valueOf(10), PUBLIC_KEY);
+    assertThat(getWatermark(VALIDATOR_ID).getSlot()).isEqualTo(UInt64.valueOf(3));
+
+    final InputStream input = createInputDataWith(List.of(6, 50), emptyList());
+    slashingProtection.importData(input);
+    assertThat(getWatermark(VALIDATOR_ID).getSlot()).isEqualTo(UInt64.valueOf(3));
+  }
+
+  @Test
+  public void watermarkDoesNotMoveIfAnyImportedBlockIsLessThanPriorMax()
+      throws JsonProcessingException {
+    final UInt64 initialBlockSlot = UInt64.valueOf(3);
+    insertValidator(PUBLIC_KEY, VALIDATOR_ID);
+    slashingProtection.registerValidators(List.of(PUBLIC_KEY));
+    insertBlockAt(initialBlockSlot, PUBLIC_KEY);
+    insertBlockAt(UInt64.valueOf(10), PUBLIC_KEY); // max = 10, watermark = 3 (As was first).
+    assertThat(getWatermark(VALIDATOR_ID).getSlot()).isEqualTo(initialBlockSlot);
+
+    final InputStream input = createInputDataWith(List.of(20, 9), emptyList());
+    slashingProtection.importData(input);
+    assertThat(getWatermark(VALIDATOR_ID).getSlot()).isEqualTo(UInt64.valueOf(3));
+  }
+
+  @Test
+  public void importSetsAttestationWatermarkToMinimalValuesIfNoneExist()
+      throws JsonProcessingException {
+    final InputStream input =
+        createInputDataWith(
+            emptyList(), List.of(new ImmutablePair<>(3, 4), new ImmutablePair<>(2, 5)));
+    slashingProtection.importData(input);
+    assertThat(getWatermark(VALIDATOR_ID))
+        .isEqualToComparingFieldByField(
+            new SigningWatermark(VALIDATOR_ID, null, UInt64.valueOf(2), UInt64.valueOf(4)));
+  }
+
+  @Test
+  public void attestationWatermarksUpdatedIfImportHasLargerValuesThanMaxExistingAttestation()
```

Comment:
What happens if only one if the values is greater will the watermark still be updated. i.e. import({8, 4}) what would the watermark be?

Reply:
in that case, we only update 1 of the watermarks - will create a test fo rit.